pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -236,11 +236,34 @@ func (az *Cloud) getSecurityRuleName(service *v1.Service, port v1.ServicePort, s
 		safePrefix := strings.Replace(sourceAddrPrefix, "/", "_", -1)
 		return fmt.Sprintf("shared-%s-%d-%s", port.Protocol, port.Port, safePrefix)
 	}
+	if sourceAddrPrefix == "" {
+		rulePrefix := az.getRulePrefix(service)
+		return fmt.Sprintf("%s-%s-%d", rulePrefix, port.Protocol, port.Port)
+	}
+	// ensure backward compatibility
 	safePrefix := strings.Replace(sourceAddrPrefix, "/", "_", -1)
 	rulePrefix := az.getRulePrefix(service)
 	return fmt.Sprintf("%s-%s-%d-%s", rulePrefix, port.Protocol, port.Port, safePrefix)
 }
 
+func (az *Cloud) getSourceAddressPrefix(sources []string) (string, []string, error) {
```

Comment:
`getSourceAddressPrefix` does not use az member field, it's better remove `(az *Cloud)` and also add a unit test for this func